src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1373,8 +1373,8 @@ public static string HexEscape(char character)
             return string.Create(3, character, (Span<char> chars, char c) =>
             {
                 chars[0] = '%';
-                chars[1] = UriHelper.s_hexUpperChars[(c & 0xf0) >> 4];
-                chars[2] = UriHelper.s_hexUpperChars[c & 0xf];
+                chars[1] = (char)UriHelper.HexUpperChars[(c & 0xf0) >> 4];
+                chars[2] = (char)UriHelper.HexUpperChars[c & 0xf];
```

Comment:
You could flip the order (put index 2 on top) to elide two bound-checks? (a real nano optimization)

Reply:
Thanks.  We can do that separately if it proves to produce meaningful improvements.

Reply:
We do something similar to this in a few places already. [See here.](https://source.dot.net/#System.Private.CoreLib/shared/System/Buffers/Text/Utf8Formatter/FormattingHelpers.cs.html,8ac7553fe384309f,references) That gives generally better performance than a lookup table anyway, and it should be possible to extend the same general pattern to chat in addition to byte.